File: javascript/lists/csv.js
function(head, req) {
  var row;
  start({
    "headers": {
      "Content-Type": "text/plain"
     }
  });

  send("key,rate\n");
  pr=null
  while(row = getRow()) {
	var cn=row.key[0];
	var nm=row.key[1];
	var id=row.key[2];
	var rt=row.value;

	if (req.query.hasOwnProperty("s"))
		if (id<req.query.s.replace(/"/g,""))
			continue;

	if (req.query.hasOwnProperty("e"))
		if (id>req.query.e.replace(/"/g,""))
			continue;

	if (pr==id)
		continue;

	pr=id

	send(cn+"/"+nm+","+id+","+rt+ "\n");
  }

}

File: erlang/lists/csv.erl
fun(Head, {Req}) ->
  Fun = fun({Row}, Acc) ->
    Id = couch_util:get_value(<<"id">>, Row),
    [A|[B|[C]]]= couch_util:get_value(<<"key">>, Row),
    Value=couch_util:get_value(<<"value">>, Row),
	Send([A,<<"/">>,B,<<",">>,C,<<",">>,integer_to_binary(Value),<<"\n">>]),
    {ok,Id}
  end,
  FoldRows(Fun, nil),
  ""
end.
